### ISSUE TITLE
Profiler is not reset after calling trainer.tune() with auto_lr_find=True

### ISSUE THREAD

**Issue by jonasricker** [Lightning-AI/lightning#4512]
## 🐛 Bug

When using `SimpleProfiler` together with the Learning Rate Finder, only the timings from the call to `train` within the Learning Rate Finder are logged. For the actual training, no timings are recorded.

## Please reproduce using [the BoringModel and post here]
https://colab.research.google.com/drive/1FUOW-A7gJk1fuR7ODQdMSCQkrkgamtru?usp=sharing

### Expected behavior
Profiler is reset after `trainer.tune()` is called OR the profiler is not active at all during tuning.

### Environment
* CUDA:
	- GPU:
		- Tesla T4
	- available:         True
	- version:           10.1
* Packages:
	- numpy:             1.18.5
	- pyTorch_debug:     False
	- pyTorch_version:   1.6.0+cu101
	- pytorch-lightning: 0.10.0
	- tqdm:              4.41.1
* System:
	- OS:                Linux
	- architecture:
		- 64bit
		- 
	- processor:         x86_64
	- python:            3.6.9
	- version:           #1 SMP Thu Jul 23 08:00:38 PDT 2020

### Additional context
Beside this issue, I'm wondering if calling the LR Finder manually like it's suggested [here](https://pytorch-lightning.readthedocs.io/en/latest/lr_finder.html) (`lr_finder = trainer.tuner.lr_find(model)`) might lead to some problems compared to `trainer.tune()`. In the `Trainer` class there are several things happening before `lr_find` is called. However, as far as I know the first method is necessary to obtain e.g. the plot.

**Comment by SeanNaren** [comment]
Can confirm I can reproduce this, just need some time to fix :) Here is the script that I've built out of the provided colab:

```python
import os

from torch.utils.data import DataLoader, Dataset
import pytorch_lightning as pl
from pytorch_lightning.profiler import SimpleProfiler
import os

from torch.utils.data import DataLoader, Dataset

import pytorch_lightning as pl
from pytorch_lightning.profiler import SimpleProfiler

tmpdir = os.getcwd()


class RandomDataset(Dataset):
    def __init__(self, size, length):
        self.len = length
        self.data = torch.randn(length, size)

    def __getitem__(self, index):
        return self.data[index]

    def __len__(self):
        return self.len


import torch
from pytorch_lightning import LightningModule


class BoringModel(LightningModule):

    def __init__(self):
        super().__init__()
        self.layer = torch.nn.Linear(32, 2)
        self.lr = 0.1

    def forward(self, x):
        return self.layer(x)

    def loss(self, batch, prediction):
        # An arbitrary loss to have a loss that updates the model weights during `Trainer.fit` calls
        return torch.nn.functional.mse_loss(prediction, torch.ones_like(prediction))

    def training_step(self, batch, batch_idx):
        output = self.layer(batch)
        loss = self.loss(batch, output)
        return {"loss": loss}

    def training_step_end(self, training_step_outputs):
        return training_step_outputs

    def training_epoch_end(self, outputs) -> None:
        torch.stack([x["loss"] for x in outputs]).mean()

    def validation_step(self, batch, batch_idx):
        output = self.layer(batch)
        loss = self.loss(batch, output)
        return {"x": loss}

    def validation_epoch_end(self, outputs) -> None:
        torch.stack([x['x'] for x in outputs]).mean()

    def test_step(self, batch, batch_idx):
        output = self.layer(batch)
        loss = self.loss(batch, output)
        self.log('fake_test_acc', loss)
        return {"y": loss}

    def test_epoch_end(self, outputs) -> None:
        torch.stack([x["y"] for x in outputs]).mean()

    def configure_optimizers(self):
        optimizer = torch.optim.SGD(self.layer.parameters(), lr=self.lr)
        lr_scheduler = torch.optim.lr_scheduler.StepLR(optimizer, step_size=1)
        return [optimizer], [lr_scheduler]


def test_x(tmpdir):
    # init model
    num_samples = 10000
    model = BoringModel()
    train = RandomDataset(32, num_samples)
    train = DataLoader(train, batch_size=32)
    val = RandomDataset(32, num_samples)
    val = DataLoader(val, batch_size=32)
    test = RandomDataset(32, num_samples)
    test = DataLoader(test, batch_size=32)
    # Initialize a trainer
    trainer = pl.Trainer(
        max_epochs=1,
        progress_bar_refresh_rate=1,
        auto_lr_find=True,
        profiler=SimpleProfiler()
    )

    # determine best learning rate
    trainer.tune(model, train, val)

    # Train the model ⚡
    trainer.fit(model, train, val)


test_x(tmpdir)
```

**Comment by edenlightning** [comment]
@SkafteNicki mind taking a look?

**Comment by akihironitta** [comment]
Reopening this issue for me to check if this is still an issue.

**Comment by awaelchli** [comment]
This seems to be fixed. I used the below updated script based on @SeanNaren's and compared the output from the profiler. 
The two outputs, for tuning and for the real fit, are different and seem to contain the correct counts and timings. 

Note that the profiler always records across several stages of the trainer. Which means the second output contains both the LR finder calls and the actual fit. And if there were .test() etc. it would also contain these. But if I understand correctly, this is expected and wasn't the concern of this issue.


```py

import torch
from torch.utils.data import DataLoader, Dataset

from lightning import LightningModule, Trainer
from lightning.pytorch.profilers import SimpleProfiler
from lightning.pytorch.tuner import Tuner


class RandomDataset(Dataset):
    def __init__(self, size, length):
        self.len = length
        self.data = torch.randn(length, size)

    def __getitem__(self, index):
        return self.data[index]

    def __len__(self):
        return self.len

class BoringModel(LightningModule):

    def __init__(self):
        super().__init__()
        self.layer = torch.nn.Linear(32, 2)
        self.lr = 0.1

    def forward(self, x):
        return self.layer(x)

    def loss(self, batch, prediction):
        # An arbitrary loss to have a loss that updates the model weights during `Trainer.fit` calls
        return torch.nn.functional.mse_loss(prediction, torch.ones_like(prediction))

    def training_step(self, batch, batch_idx):
        output = self.layer(batch)
        loss = self.loss(batch, output)
        return {"loss": loss}

    def training_step_end(self, training_step_outputs):
        return training_step_outputs

    def on_train_epoch_end(self) -> None:
        # torch.stack([x["loss"] for x in outputs]).mean()
        pass

    def validation_step(self, batch, batch_idx):
        output = self.layer(batch)
        loss = self.loss(batch, output)
        return {"x": loss}

    def on_validation_epoch_end(self) -> None:
        pass

    def test_step(self, batch, batch_idx):
        output = self.layer(batch)
        loss = self.loss(batch, output)
        self.log('fake_test_acc', loss)
        return {"y": loss}

    def configure_optimizers(self):
        optimizer = torch.optim.SGD(self.layer.parameters(), lr=self.lr)
        lr_scheduler = torch.optim.lr_scheduler.StepLR(optimizer, step_size=1)
        return [optimizer], [lr_scheduler]


def run():
    # init model
    num_samples = 10000
    model = BoringModel()
    train = RandomDataset(32, num_samples)
    train = DataLoader(train, batch_size=32)
    val = RandomDataset(32, num_samples)
    val = DataLoader(val, batch_size=32)
    test = RandomDataset(32, num_samples)
    test = DataLoader(test, batch_size=32)
    # Initialize a trainer
    trainer = Trainer(
        max_epochs=1,
        # progress_bar_refresh_rate=1,
        # auto_lr_find=True,
        profiler=SimpleProfiler()
    )

    tuner = Tuner(trainer)
    # determine best learning rate
    tuner.lr_find(model, train, val)

    # Train the model ⚡
    trainer.fit(model, train, val)


run()

```
